src/test/daml/Daml/Finance/Instrument/Equity/Test/DivOption.daml

```diff
@@ -0,0 +1,225 @@
+import Daml.Finance.Interface.Instrument.Option.Dividend.Types (ElectionTypeEnum(..))
+import Daml.Finance.Interface.Lifecycle.Rule.Claim qualified as Claim (ClaimEffect(..), I)
+import Daml.Finance.Interface.Lifecycle.Rule.Lifecycle qualified as Lifecycle (Evolve(..), I)
+import Daml.Finance.Interface.Settlement.Batch qualified as Batch (Settle(..))
+import Daml.Finance.Interface.Settlement.Instruction qualified as Instruction (Allocate(..), Approve(..))
+import Daml.Finance.Interface.Settlement.Types (Allocation(..), Approval(..))
+import Daml.Finance.Interface.Types.Common.Types (Id(..), Quantity(..))
+import Daml.Finance.Interface.Util.Common (qty)
+import Daml.Finance.Lifecycle.Rule.Claim qualified as Claim (Rule(..))
+import Daml.Finance.Lifecycle.Rule.Distribution qualified as Distribution (Rule(..))
+import Daml.Finance.Settlement.Factory (Factory(..))
+import Daml.Finance.Settlement.RouteProvider.SingleCustodian (SingleCustodian(..))
+import Daml.Finance.Test.Util.Account qualified as Account (ControlledBy(..), createAccount, createFactory, credit)
+import Daml.Finance.Test.Util.Common (createParties)
+import Daml.Finance.Test.Util.Holding qualified as Holding (verifyOwnerAndAmountOfHolding)
+import Daml.Finance.Test.Util.Instrument (originate)
+import Daml.Finance.Test.Util.Instrument qualified as Instrument (submitExerciseInterfaceByKeyCmd)
+import Daml.Script
+
+-- Issue dividend option and do an election: lifecycling and settlement
+run : Script ()
+run = script do
+  -----------------------------
+  -- Issue a dividend option --
+  -----------------------------
+
+  -- Create parties
+  [cb, issuer, custodian, investor, publicParty] <-
+    createParties ["CentralBank", "Issuer", "Custodian", "Investor", "PublicParty"]
+
+  -- Create holding and account factory
+  let pp = [("PublicParty", singleton publicParty)]
+  accountFactoryCid <- toInterfaceContractId <$> Account.createFactory issuer pp
+  holdingFactoryCid <- toInterfaceContractId <$> submit issuer do
+    createCmd Fungible.Factory with provider = issuer; observers = M.fromList pp
+
+  -- Create accounts
+  investorAccount <-
+    Account.createAccount "Investor Account" [publicParty] accountFactoryCid holdingFactoryCid []
+    Account.Owner custodian investor
+  issuerAccount <-
+    Account.createAccount "Issuer Account" [publicParty] accountFactoryCid holdingFactoryCid []
+    Account.Owner custodian issuer
+
+  let
+    expiryDate = date 2019 May 15
+
+  -- Originate instruments
+  now <- getTime
+  cashInstrument <- originate cb cb "USD" "United States Dollar" pp now
+  eurInstrument <- originate cb cb "EUR" "Euro"  pp now
+  cumEquityInstrument <- originateEquity issuer issuer "EQUITY-INST-1" "0" "ABC" pp now
+  exEquityInstrument <- originateEquity issuer issuer "EQUITY-INST-1" "1" "ABC" [] now
+
+  let
+    divAmountCash = 0.02
+    divAmountFxCash = 0.025
+    divAmountShares = 0.03
+    cashQuantity = Quantity cashInstrument divAmountCash
```

Comment:
fyi, we also have a `qty` builder in `D.F.I.U.Common`

Reply:
I switched to `qty` instead.